src/cascadia/TerminalControl/ControlCore.cpp

```diff
@@ -974,24 +985,77 @@ namespace winrt::Microsoft::Terminal::Control::implementation
 
     Control::IControlSettings ControlCore::Settings()
     {
-        return *_settings;
+        return _settings;
     }
 
     Control::IControlAppearance ControlCore::FocusedAppearance() const
     {
-        return *_settings->FocusedAppearance();
+        return _settings;
     }
 
     Control::IControlAppearance ControlCore::UnfocusedAppearance() const
     {
-        return *_settings->UnfocusedAppearance();
+        return _unfocusedAppearance;
+    }
+
+    void ControlCore::ApplyPreviewColorScheme(const Core::ICoreScheme& scheme)
+    {
+        const auto lock = _terminal->LockForReading();
+        auto& renderSettings = _terminal->GetRenderSettings();
+        if (!_stashedColorScheme)
+        {
+            // There is a contract between Apply- and ResetPreviewColorScheme as to the layout of `_stashedColorScheme`.
+            // It is not intended for use outside these two functions.
+            // It contains the every color in the table (some of which will be overwritten by the color scheme),
+            // followed by the two indices for the foreground/background alias (set by DECAC, and which we overwrite
+            // when applying a color scheme) stored as COLORREF for convenience.
+            // We store the entire table because TerminalCore stores the default foreground, background, frame colors,
+            // cursor color and selection color at the _end_ of the palette. This is an easy way to collect them all and
+            // restore them all as well.

```

Comment:
Is there a reason you don't declare a
```cpp
struct StashedColorScheme {
    std::array<COLORREF, TextColor::TABLE_SIZE> scheme;
    COLORREF foreground;
    COLORREF background;
};
```

You can then probably just use `operator=` with the std::array.

Reply:
done thanks!